nav2_behavior_tree/src/generate_nav2_tree_nodes_xml.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 Davide Faconti
```

Comment:
I assume this is automatically executed on build?

I think this filename is misleading, its not generating the XML but just storing the index of default Nav2-included BT nodes. How about `generate_nav2_node_list.cpp`?

Reply:
> I think this filename is misleading, its not generating the XML

it is literally creating an XML with `std::ofstream` ... but I can change the name, of course.

> I assume this is automatically executed on build? 

No, it is not. The developer modifying a BT Node should run it and overwrite the previous XML with the new one.

But right now it is even worse, because we are asking to write the XML by hand and there is no guaranty of consistency (as a demonstration, right now it is WRONG  in the **main** branch)

Reply:
>  this workflow would add extra steps

That step ("double check if **nav2_tree_node.xml** is still valid") is there, but people just forget to do it.
This approach lowers the effort.

Maybe I can find a way to run it automatically, I don't know...

Reply:
> No, it is not. The developer modifying a BT Node should run it and overwrite the previous XML with the new one.

Oh, then this should be removed. You cannot expect the CI system or users to run some script just to get their systems working or unit tests compiling. This doesn't work for CI nor a reasonable developer workflow

Reply:
Sorry, there is a misunderstanding from my side and maybe I answered something wrong
**CI, user code and unit test will work just fine!!!**
The file that is autogenerated is `plugin_list.hpp`. That is everything you need.

What I am talking about is the file **nav2_tree_node.xml** , that only Groot2 requires.

Reply:
Oh ok. So this does generate the index that the `plugins_list` needs for the BT test + BT navigator. Cool then this works.

Now you got confused by this utility's name :wink: `generate_nav2_node_list.cpp` FTW